### ISSUE TITLE
Reader for Array[Float] returns `Infinity` value for `-1234567890123456789e55`

### ISSUE THREAD

**Issue by plokhotnyuk** [ngs-doo/dsl-json#123]
Code to reproduce:
```
val r = new DslJson[Any](new DslJson.Settings[Any]()).newReader()
val bs = "[-1234567890123456789e55]".getBytes
r.process(bs, bs.length)
r.read()
println(NumberConverter.FLOAT_ARRAY_READER.read(r)(0))
```

**Comment by plokhotnyuk** [comment]
Also it returns positive value for some negative like `-1011169427`

**Comment by zapov** [comment]
Tnx. I've fixed the first problem in: https://github.com/ngs-doo/dsl-json/commit/7cce634e63a326efe37e56a865ad881a0ec1dff8
Wasn't able to reproduce the second one with this number (but I think it should be the same case - when there are more than 18 digits it goes into same path and invokes Java parsing). Negation should work correctly for that now.

**Comment by plokhotnyuk** [comment]
Now getting `Infinity` when parsing `[0E+75]` as Array[Float].

**Comment by zapov** [comment]
More special cases ;( Fixed in https://github.com/ngs-doo/dsl-json/commit/23a6c25888ab9d09a515c594570e69d58fd29ab2

**Comment by zapov** [comment]
I've released v1.9.1 which has relevant fixes
